schemas/codify.xsd

```diff
@@ -1,50 +1,248 @@
+        </xs:complexType>
+    </xs:element>
+    <!-- types -->
+    <xs:complexType name="anyType">
+        <xs:complexContent>
+            <xs:extension base="xs:anyType">
+                <xs:attribute name="warning" type="falseOnlyType" use="optional"/>
+            </xs:extension>
+        </xs:complexContent>
     </xs:complexType>
-  </xs:element>
-  <xs:attribute name="applicability" type="xs:string"/>
-  <xs:attribute name="doc" type="xs:string"/>
-  <xs:attribute name="expire" type="xs:string"/>
-  <xs:attribute name="instruction" type="xs:string"/>
-  <xs:attribute name="path" type="xs:string"/>
-  <xs:attribute name="value" type="xs:string"/>
-  <xs:attributeGroup name="allAttributes">
-    <xs:attribute ref="applicability" use="optional" />
-    <xs:attribute ref="doc" use="optional" />
-    <xs:attribute ref="expire" use="optional"/>
-    <xs:attribute ref="instruction" use="optional" />
-    <xs:attribute ref="path" use="optional" />
-    <xs:attribute ref="value" use="optional" />
-  </xs:attributeGroup>
-  <xs:attributeGroup name="targetAttributes">
-    <xs:attribute ref="doc" use="optional" />
-    <xs:attribute ref="path" use="optional" />
-  </xs:attributeGroup>
+    <xs:simpleType name="falseOnlyType">
+        <xs:restriction base="xs:string">
+            <xs:pattern value="false" />
+        </xs:restriction>
+    </xs:simpleType>
+    <xs:complexType name="insertType">
+        <xs:sequence>
+            <xs:choice minOccurs="0" maxOccurs="unbounded">
+                <xs:element ref="create-sub-container"/>
+                <xs:element ref="insert"/>
+                <xs:element ref="recodify-section"/>
+                <xs:element ref="lib:container"/>
+                <xs:element ref="lib:section"/>
+                <xs:element ref="lib:para"/>
+                <xs:element ref="lib:text"/>
+                <xs:element ref="lib:subsection"/>
+                <xs:element ref="lib:toc"/>
+            </xs:choice>
+        </xs:sequence>
+    </xs:complexType>
+    <!-- attributes -->
+    <xs:attribute name="applicability" type="xs:string"/>
+    <xs:attribute name="doc" type="xs:string"/>
+    <xs:attribute name="expire" type="xs:string"/>
+    <xs:attribute name="instruction" type="xs:string"/>
+    <xs:attribute name="path" type="xs:string"/>
+    <xs:attribute name="warning" type="falseOnlyType"/>
+    <xs:attribute name="value" type="xs:string"/>
+    <!-- attribute groups -->
+    <xs:attributeGroup name="allAttributes">
+        <xs:attribute ref="applicability" use="optional" />
+        <xs:attribute ref="doc" use="optional" />
+        <xs:attribute ref="expire" use="optional"/>
+        <xs:attribute ref="instruction" use="optional" />
+        <xs:attribute ref="path" use="optional" />
+        <xs:attribute ref="value" use="optional" />
+        <xs:attribute name="warning" type="falseOnlyType" use="optional"/>
+    </xs:attributeGroup>
+    <xs:attributeGroup name="historyAttributes">
```

Comment:
Should this be on all of the `<codify:` elements?

Reply:
maybe should be changed to something like sharedAttributes, and should add warning to this list.